core/templates/pages/contributor-dashboard-page/services/contribution-opportunities.service.ts

```diff
@@ -17,115 +17,151 @@
+    this.ngbModal.open({
+      templateUrl: this.urlInterpolationService.getDirectiveTemplateUrl(
+        '/pages/community-dashboard-page/modal-templates/' +
+                'login-required-modal.directive.html'),
+      backdrop: 'static',
+      controller: [
+        '$controller', '$scope', '$uibModalInstance',
+        function($controller, $scope, $uibModalInstance) {
+          $controller('ConfirmOrCancelModalController', {
+            $scope: $scope,
+            $uibModalInstance: $uibModalInstance
           });
-    };
-
-    var showRequiresLoginModal = function(argument) {
-      $uibModal.open({
-        templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
-          '/pages/community-dashboard-page/modal-templates/' +
-          'login-required-modal.directive.html'),
-        backdrop: 'static',
-        controller: [
-          '$controller', '$scope', '$uibModalInstance',
-          function($controller, $scope, $uibModalInstance) {
-            $controller('ConfirmOrCancelModalController', {
-              $scope: $scope,
-              $uibModalInstance: $uibModalInstance
-            });
-          }]
-      }).result.then(function() {}, function() {
-        // Note to developers:
-        // This callback is triggered when the Cancel button is clicked.
-        // No further action is needed.
-      });
-    };
-
-    return {
-      getSkillOpportunitiesAsync: async function() {
-        return _getSkillOpportunities('');
-      },
-      getTranslationOpportunitiesAsync: async function(languageCode) {
-        return _getTranslationOpportunities(languageCode, '');
-      },
-      getVoiceoverOpportunities: async function(languageCode) {
-        return _getVoiceoverOpportunities(languageCode, '');
-      },
-      getMoreSkillOpportunitiesAsync: async function() {
-        if (moreSkillOpportunitiesAvailable) {
-          return _getSkillOpportunities(skillOpportunitiesCursor);
-        }
-      },
-      getMoreTranslationOpportunitiesAsync: async function(languageCode) {
-        if (moreTranslationOpportunitiesAvailable) {
-          return _getTranslationOpportunities(
-            languageCode, translationOpportunitiesCursor);
-        }
-      },
-      getMoreVoiceoverOpportunities: async function(languageCode) {
-        if (moreVoiceoverOpportunitiesAvailable) {
-          return _getVoiceoverOpportunities(
-            languageCode, voiceoverOpportunitiesCursor);
-        }
-      },
-      showRequiresLoginModal: showRequiresLoginModal,
-      reloadOpportunitiesEventEmitter: reloadOpportunitiesEventEmitter,
-      removeOpportunitiesEventEmitter: removeOpportunitiesEventEmitter,
-    };
-  }]);
+        }]
+    }).result.then(function() {
+    }, function() {
+      // Note to developers:
+      // This callback is triggered when the Cancel button is clicked.
+      // No further action is needed.
+    });
+  }
+
+  async getSkillOpportunitiesAsync(): Promise<Object> {
```

Comment:
Can you search for the use of this function everywhere in the codebase and add $rootscope.$applyAsync() in the required places. Do this for all the other public functions. Also, Promise<Object> is not a good way to represent types. Try and find out the correct type (I know the backend API service has Promise<Object> as the type but try and find out what the type is.

Reply:
@srijanreddy98 I'm not getting the use of `$rootScope.$applyAsync`. 

`ctrl.loadOpportunities = function() {
        return ContributionOpportunitiesService
          .getTranslationOpportunitiesAsync(
            TranslationLanguageService.getActiveLanguageCode()).then(
            getPresentableOpportunitiesData);
      };` Where do I apply it in the above function??

Reply:
```
ctrl.loadOpportunities = function() { 
  return ContributionOpportunitiesService.getTranslationOpportunitiesAsync( 
    TranslationLanguageService.getActiveLanguageCode()).then( 
      translationOpportunities => {
        getPresentableOpportunitiesData(translationOpportunities);
        $rootScope.$applyAsync();
     }
  );
}
```